vllm/model_executor/models/qwen2.py

```diff
@@ -349,7 +349,7 @@ def load_weights(self,
             ("gate_up_proj", "gate_proj", 0),
             ("gate_up_proj", "up_proj", 1),
         ]
-        params_dict = dict(self.named_parameters())
+        params_dict = dict(self.named_parameters(remove_duplicate=False))
```

Comment:
Could you explain why this fixes the bug? Sorry I don’t have context for this.

Reply:
```
        if config.tie_word_embeddings:
            self.lm_head_weight = self.model.embed_tokens.weight
        else:
            self.lm_head = ParallelLMHead(config.vocab_size,
                                          config.hidden_size)
            self.lm_head_weight = self.lm_head.weight
```
Both have two same parameters which will be removed in `self.named_parameters(remove_duplicate=True)`.
So If we use another temp parameter `self.lm_head_weight`, we need use `remove_duplicate=False` like other models.

Reply:
I see. Thanks! TIL